crates/typst/src/foundations/label.rs

```diff
@@ -78,3 +78,7 @@ impl From<Label> for PicoStr {
 
 /// Indicates that an element cannot be labelled.
 pub trait Unlabellable {}
+
+pub(crate) fn is_valid_in_label_literal(id: &str) -> bool {
+    !id.is_empty() && id.chars().all(typst_syntax::is_valid_in_label)
+}
```

Comment:
It seems that this function is simple and only used once in `CastInfo::error`.

I'm not sure if this is over-encapsulation.

Reply:
I think having this is fine, too, since the implementation isn’t exactly trivial, but I’ve thought about whether to put it here or in `typst_syntax`.

Reply:
The naming is a bit tricky. Right now the naming doesn't make the difference clear.

We could go with
- `typst_syntax::is_valid_in_label` -> `typst_syntax::is_valid_char_in_label_literal`
- `is_valid_in_label_literal` -> `typst_syntax::is_valid_in_label_literal`

Or we could go with @Coekjan's reasoning and have only the char-version as `typst_syntax::is_valid_in_label_literal`. Makes the naming easier.

Reply:
I opted for:

* `typst_syntax::is_valid_in_label` → `typst_syntax::is_valid_in_label_literal`
* `is_valid_in_label_literal` → `typst_syntax::is_valid_label_literal_id`